### PR TITLE
Propagate xattr from UfsStatus to Inode

### PR DESCRIPTION
**Pull request by ZacBlanco**
This change aims to allow the calls to the UFS from the
FileSystemMaster to propagate into the inode tree. This will
allow developers in the future to leverage any xattr that are
returned from the UFS implementations in order to make
optimizations or other assumptions about the state of a node
past the standard set of fields.

Most of these changes are mechanical, it's just a matter of
finding exactly where to set the xattr and where not to.

For now, the only usecase for setting xattr is during sync or
loadMetadata calls. The implementation adds the xattr to the
create*Context so that the inode tree can utilize set the xattr
on inodes from within the inode tree class.

### CODE REVIEW

File: core/server/master/src/main/java/alluxio/master/file/contexts/CreatePathContext.java [hunk 1]
```diff
@@ -258,6 +261,22 @@ public K setMetadataLoad(boolean metadataLoad) {
     return getThis();
   }
 
+  /**
+   * @return extended attributes on this context
+   */
+  public Map<String, byte[]> getXAttr() {
```
Comment (gpang) [inline core/server/master/src/main/java/alluxio/master/file/contexts/CreatePathContext.java]: does this need the `@Nullable` annotation?

File: core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java [hunk 2]
```diff
@@ -668,6 +670,7 @@ public InodeDirectory getRoot() {
           mState.applyAndJournal(rpcContext, UpdateInodeEntry.newBuilder()
               .setId(currentId)
               .setLastModificationTimeMs(context.getOperationTimeMs())
+              .putAllXAttr(CommonUtils.convertToByteString(context.getXAttr()))
```
Comment (gpang) [inline core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java]: Can `context.getXAttr()` be `null`?

**Comment by gpang** [comment]
Jenkins, test this please

**Comment by gpang** [comment]
alluxio-bot, merge this please

**Comment by calvinjia** [comment]
alluxio-bot, cherry-pick this to branch-2.0 please.

**Pull request merged**